IronFrame.Test/Utilities/FirewallManagerTests.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NetFwTypeLib;
+using Xunit;
+
+namespace IronFrame.Utilities
+{
+    public class FirewallManagerTests
+    {
+        private const string Username = "Administrator";
+        readonly FirewallManager manager = new FirewallManager();
+        readonly INetFwPolicy2 firewallPolicy =
+               (INetFwPolicy2)Activator.CreateInstance(Type.GetTypeFromProgID("HNetCfg.FwPolicy2"));
+
+        public void CheckCommonRuleProperties(INetFwRule3 rule)
+        {
+            Assert.NotNull(rule);
+            Assert.Equal(NET_FW_ACTION_.NET_FW_ACTION_ALLOW, rule.Action);
+            Assert.Equal(NET_FW_RULE_DIRECTION_.NET_FW_RULE_DIR_OUT, rule.Direction);
+            Assert.Equal(true, rule.Enabled);
+            Assert.Equal(FirewallManager.GetFormattedLocalUserSid(Username), rule.LocalUserAuthorizedList);
+        }
+
+        [Collection("Firewall Test Collection")]
+        public class TcpTest : FirewallManagerTests
+        {
+            [FactAdminRequired]
+            public void CreateFirewallRuleWithEmptySpec()
+            {
+                try
+                {
+                    manager.CreateFirewallRule(Username, new FirewallRuleSpec
+                    {
+                        Protocol =  Protocol.Tcp,
+                    });
+                    var rule = (INetFwRule3)firewallPolicy.Rules.Item(Username);
+                    CheckCommonRuleProperties(rule);
+                    Assert.Equal(rule.Protocol, (int)NET_FW_IP_PROTOCOL_.NET_FW_IP_PROTOCOL_TCP);
+                    Assert.Equal("*", rule.RemoteAddresses);
+                    Assert.Equal("*", rule.RemotePorts);
+                }
+                finally
+                {
+                    manager.RemoveAllFirewallRules(Username);
```

Comment:
This "Dispose" logic can be handled by the test class.  If the test implements `IDisposable` then you can remove the rules in the Dispose method.

The test class is created and disposed for each test.  This allows you to have common cleanup for each test.